cloudpickle/cloudpickle_fast.py

```diff
@@ -0,0 +1,554 @@
+        if func.__closure__ is not None else ()
+    )
+
+    # Extract currently-imported submodules used by func. Storing these modules
+    # in a smoke _cloudpickle_subimports attribute of the object's state will
+    # trigger the side effect of importing these modules at unpickling time
+    # (which is necessary for func to work correctly once depickled)
+    slotstate["_cloudpickle_submodules"] = _find_imported_submodules(
+        func.__code__, itertools.chain(f_globals.values(), closure_values))
+    slotstate["__globals__"] = f_globals
+
+    state = func.__dict__
+    return state, slotstate
+
+
+def _class_getstate(obj):
+    clsdict = _extract_class_dict(obj)
+    clsdict.pop('__weakref__', None)
+    clsdict.pop('__doc__', None)  # present in the reconstructor args
+
+    # For ABCMeta in python3.7+, remove _abc_impl as it is not picklable.
+    # This is a fix which breaks the cache but this only makes the first
+    # calls to issubclass slower.
+    if "_abc_impl" in clsdict:
+        (registry, _, _, _) = abc._get_dump(obj)
+        clsdict["_abc_impl"] = [subclass_weakref()
+                                for subclass_weakref in registry]
+    if hasattr(obj, "__slots__"):
+        # pickle string length optimization: member descriptors of obj are
+        # created automatically from obj's __slots__ attribute, no need to
+        # save them in obj's state
+        if isinstance(obj.__slots__, string_types):
+            clsdict.pop(obj.__slots__)
+        else:
+            for k in obj.__slots__:
+                clsdict.pop(k, None)
+
+    clsdict.pop('__dict__', None)  # unpicklable property object
+
+    return (clsdict, {})
+
+
+def _enum_getstate(obj):
+    clsdict, slotstate = _class_getstate(obj)
+
+    members = dict((e.name, e.value) for e in obj)
+    # Cleanup the clsdict that will be passed to _rehydrate_skeleton_class:
+    # Those attributes are already handled by the metaclass.
+    for attrname in ["_generate_next_value_", "_member_names_",
+                     "_member_map_", "_member_type_",
+                     "_value2member_map_"]:
+        clsdict.pop(attrname, None)
+    for member in members:
+        clsdict.pop(member)
+        # Special handling of Enum subclasses
+    return clsdict, slotstate
+
+
+# COLLECTIONS OF OBJECTS REDUCERS
+# -------------------------------
+# A reducer is a function taking a single argument (obj), and that returns a
+# tuple with all the necessary data to re-construct obj. Apart from a few
+# exceptions (list, dict, bytes, int, etc.), a reducer is necessary to
+# correctly pickle an object.
+# While many built-in objects (Exceptions objects, instances of the "object"
+# class, etc), are shipped with their own built-in reducer (invoked using
+# obj.__reduce__), some do not. The following methods were created to "fill
+# these holes".
+
+def _code_reduce(obj):
+    """codeobject reducer"""
+    if hasattr(obj, "co_posonlyargcount"):  # pragma: no branch
```

Comment:
This is always the case in Python 3.8+ right? The else branch is not covered by the tests:

https://codecov.io/gh/cloudpipe/cloudpickle/compare/f3c3aeae28a650f3ed0ea5f8516e2f090ce44536...3b879044a2e4d46350baf9e77a39da63355d4ae7/diff

It might be useful if we decide to make cloudpickle leverage `pickle5` as an optional dependency and support fast pickling for python 3.6+ in that case, but as of now I think we can remove the unreachable case.

Reply:
OK.

Reply:
Yes we don't care about alpha releases anymore now that the 3.8b1 is out.